### ISSUE TITLE
[Discussion] Should there be a cutoff date for inclusion in the timeline?

### ISSUE THREAD

**Issue by juliaferraioli** [juliaferraioli/opensource-timeline#15]
In light of PR #14, should we have a date that all events need to fall after? We can go way far back into the weeds of the history of computing, but at some point then we make this a history of computing timeline instead of an open source one.

I'm in favor of a cutoff date, tentatively 1940 or so, but would like to hear other opinions.

**Comment by joshuabezaleel** [comment]
How about instead a cutoff date in a specific year or any other time format, the cutoff is a specific event where the events coming after that event along with the event brings the significance of the story in the timeline of open source to differentiate it with the computing ones? The current ones that we have at the `.csv` referring to the event of `Software shared as public domain` is a good starter but I would say the more specific the better. Perhaps maybe the first public repository, or the first issue opened/pull request sent. 

Anyway really appreciate the initiative of open source timeline and open source stories, thank you for starting this, Julia! Will be happy to help if I have some free time onwards.

**Comment by dneary** [comment]
I would say there's a "before software was developed and sold separate from hardware" line. That era started with the creation of Computer Usage Company in 1955, probably, although software was still custom built. The first software product might have been an ALGOL or FORTRAN compiler in the early 1960s (Maybe the JOVIAL compiler by CSC?). Everything after that has the decision around how software should be shared (source only, source + binary, binary only) and the conditions of sharing.

Also, the "Computer Software Copyright Act" in 1980 marked a huge turning point in source code being considered copyrightable - there was a "before CSCA" and "after CSCA". Which coincidentally is roughly aligned with the publication of "An Open Letter to Computer Hobbyists".

**Comment by badnetmask** [comment]
> How about instead a cutoff date in a specific year or any other time format, the cutoff is a specific event where the events coming after that event along with the event brings the significance of the story in the timeline of open source to differentiate it with the computing ones? The current ones that we have at the `.csv` referring to the event of `Software shared as public domain` is a good starter but I would say the more specific the better. Perhaps maybe the first public repository, or the first issue opened/pull request sent. 
> 

I came here to say mostly what you already said: events that were significant to the birth of the open source model.

Just one comment though: I *believe* public repos (like, over the internet) came a long while after the idea of open source, so I would be careful calling that "the line".

**Comment by juliaferraioli** [comment]
> the cutoff is a specific event where the events coming after that event along with the event brings the significance of the story in the timeline of open source to differentiate it with the computing ones?

Yeah, I think that's fairly reasonable. Initially I had an `era` `eventType` in the schema, which would cover some of the broader cultural happenings, but it was suggested that it would be confusing.

> Also, the "Computer Software Copyright Act" in 1980 marked a huge turning point in source code being considered copyrightable - there was a "before CSCA" and "after CSCA". Which coincidentally is roughly aligned with the publication of "An Open Letter to Computer Hobbyists".

Absolutely. That's on my list to add (unless someone gets to it before me, hint hint).

> "before software was developed and sold separate from hardware" line

Oooh. That's a bit harder to articulate and screen for, but I like this. The good thing is that there *is* a date proxy.

**Comment by dneary** [comment]
> > Also, the "Computer Software Copyright Act" in 1980 marked a huge turning point in source code being considered copyrightable - there was a "before CSCA" and "after CSCA". Which coincidentally is roughly aligned with the publication of "An Open Letter to Computer Hobbyists".
> 
> Absolutely. That's on my list to add (unless someone gets to it before me, hint hint).

:-D

> > "before software was developed and sold separate from hardware" line
> 
> Oooh. That's a bit harder to articulate and screen for, but I like this. The good thing is that there _is_ a date proxy.

Fully enough, "the first software product" probably happened around the same time as the first open source - it would have been an ALGOL 58 or ALGOL 60 compiler - and there were several ALGOL60 compilers that made the rounds of universities that were freely shared (although of course there were no licenses back then).

**Comment by ShaneCurcuru** [comment]
What is the purpose of the timeline?

I'd vote for "important events that shaped how open source software evolved".  Thus there'd be a few key events from 50s - 70s, things that people may have heard of (ARPANET, copyright law change), and could see why those events are actually important today.

**Comment by juliaferraioli** [comment]
> What is the purpose of the timeline?

Really, it's to get a collection of important events, with references and metadata into a structured format. And to get the knowledge out of my head so I can reclaim the space 😆 

I had originally put together [this timeline](https://time.graphics/line/569930) for a talk and it was a pain drawing from a bunch of sources. I figure folks might want a more comprehensive, machine-readable timeline. And if they don't, **I** do!

> I'd vote for "important events that shaped how open source software evolved". Thus there'd be a few key events from 50s - 70s, things that people may have heard of (ARPANET, copyright law change), and could see why those events are actually important today.

That works for me, but I also want to acknowledge that it'll likely be a moving target as to what counts as important.
